### PR TITLE
Custom Functions batching files/folders

### PR DESCRIPTION
**Pull request by mscharlock**

### CODE REVIEW

File: Excel-custom-functions/Batching/samplewithtodos.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+//Custom Functions Batching Sample with TODO items
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/samplewithtodos.js]: Sentence case this.

File: Excel-custom-functions/Batching/samplewithtodos.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+//TODO #1: Create custom functions for adding and multiplying numbers.
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/samplewithtodos.js]: Remove gerunds whenever you can. Including code samples! "Create custom functions that perform addition and multiplication calculations" -or- "Create custom functions to add and multiply numbers."

File: Excel-custom-functions/Batching/samplewithtodos.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    // schedule it after a certain timeout, e.g. 2 sec.
+    
+    //Return the promise for this invocation.
+    return promise;
+  }
+  
+  var _batch = [];
+  var _isBatchedRequestScheduled = false;
+  
+  // TODO #3: Write a function which creates a copy of the batch array, empty the batch array,
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/samplewithtodos.js]: Need verb tenses to match: creates, empties, creates. or create, empty, create, pick one or the other.

File: Excel-custom-functions/Batching/samplewithtodos.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  // and create a new array containing only the raw information to be processed by the API. 
+  
+  // TODO #4: In the same function for TODO #3, make the request to the API and match the values
+  // from the returned results to the corresponding invocation entry from the request batch. 
+  
+  
+  // --------------------- A public API ------------------------------
+  
+  // This function simulates the work of a remote service and is here for your reference only. Because each service
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/samplewithtodos.js]: you'll

File: Excel-custom-functions/Batching/samplewithtodos.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  // differs, you will need to modify this function appropriately to work with the service you are using.
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/samplewithtodos.js]: you're

File: Excel-custom-functions/Batching/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+* [Debugging](#debugging)
+* [IntelliSense for the JSON file in Visual Studio Code](#intellisense-for-the-json-file-in-visual-studio-code)
+* [Questions and comments](#questions-and-comments)
+* [Additional resources](#additional-resources)
+
+## Change History
+
+* Oct 27, 2017: Initial version.
+* April 23, 2018: Revised and expanded.
+* June 1, 2018: Bug fixes.
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: Add "March 20, 2019: Removed Preview" (or whatever) when this is no longer preview!

File: Excel-custom-functions/Batching/README.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+## Prerequisites
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: For SEO, change to "Prerequisites for creating Excel Custom Functions"

File: Excel-custom-functions/Batching/README.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+* Install Office 2016 for Windows and join the [Office Insider](https://products.office.com/en-us/office-insider) program. You must have Office build number 10827 or later.
+
+## To use the project
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: This heading is out of agreement with the other headings. Change to "Using the Excel custom functions project"

File: Excel-custom-functions/Batching/README.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+2. From a command prompt from within your custom functions project directory, run `npm run start` to start a localhost server instance. 
+3. Run `npm run sideload` to launch Excel and load the custom functions add-in. Additonal information on sideloading can be found at <https://aka.ms/sideload-addins>.
+4. After Excel launches, you will need to register the custom-functions add-in to work around a bug:
+    a. On the upper-left-hand side of Excel, there is a small hexagon icon with a dropdown arrow. The icon is to right of the Save icon.
+    b. Click on this dropdown arrow and then click on the Custom Functions Sample add-in to register it.
+5. Test a custom function by entering `=CONTOSO.ADD(num1, num2)` in a cell.
+6. Try the other functions in the sample: `=CONTOSO.ADDASYNC(num1, num2)`, `CONTOSO.INCREMENTVALUE(increment)`.
+7. If you make changes to the sample add-in, copy the updated files to your website, and then close and reopen Excel. If your functions are not available in Excel, re-insert the add-in using **Insert** > **My Add-ins**.
+
+## Making changes
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: Suggest "Making code changes in the custom functions project"

File: Excel-custom-functions/Batching/README.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+If you make changes to the sample functions code (in the JS file), close and reopen Excel to test them.
+
+If you change the functions metadata (in the JSON file), close Excel and delete your cache folder `Users/<user>/AppData/Local/Microsoft/Office/16.0/Wef/CustomFunctions`. Then re-insert the add-in using **Insert** > **My Add-ins**.
+
+## Debugging
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: Debugging your Excel custom function

File: Excel-custom-functions/Batching/README.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+If your add-in fails to register, [verify that SSL certificates are correctly configured](https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md) for the web server that's hosting your add-in application.
+
+If you are testing your add-in in Office 2016 desktop you can enable [runtime logging](https://docs.microsoft.com/office/dev/add-ins/testing/troubleshoot-manifest#use-runtime-logging-to-debug-your-add-in) to debug issues with your add-in's XML manifest file as well as several installation and runtime conditions.
+
+## IntelliSense for the JSON file in Visual Studio Code	
+For intelliSense to help you edit the JSON file, follow these steps:
+
+1. Open the JSON file (it has a .json extension) in Visual Studio Code.	
+2. If you are starting a new file from scratch, add the following to the top of the file:
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: Use contractions: You are should be you're (check elsewhere too, I think I noticed a couple others.)

File: Excel-custom-functions/Batching/README.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+* April 23, 2018: Revised and expanded.
+* June 1, 2018: Bug fixes.
+
+## Prerequisites
+
+* Install Office 2016 for Windows and join the [Office Insider](https://products.office.com/en-us/office-insider) program. You must have Office build number 10827 or later.
+
+## To use the project
+
+On a machine with a valid instance of an Excel Insider build installed, follow these instructions to use this custom function sample add-in:
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: change "machine" to "computer"

File: Excel-custom-functions/Batching/README.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+1. On the machine where your custom functions project is installed, follow the instructions to install the self-signed certificates (https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md) . 
+2. From a command prompt from within your custom functions project directory, run `npm run start` to start a localhost server instance. 
+3. Run `npm run sideload` to launch Excel and load the custom functions add-in. Additonal information on sideloading can be found at <https://aka.ms/sideload-addins>.
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: "For additional information about sideloading, see ..."

File: Excel-custom-functions/Batching/README.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+4. After Excel launches, you will need to register the custom-functions add-in to work around a bug:
```
Comment (lindalu-MSFT) [inline Excel-custom-functions/Batching/README.md]: you will should be "you'll"

**Comment by lindalu-MSFT** [comment]
@davidchesnut Hi David! I don't have admin perms for this repo. Do you want me to starting publishing it during my weekly publishing cadence?

**Comment by davidchesnut** [comment]
Thanks @lindalu-MSFT for now let me merge these to master, but let's follow up down the road as I think having same process is a good idea.

**Pull request merged**